tests/Network/SocketSpec.hs

```diff
@@ -95,6 +97,24 @@ spec = do
         it "converts an index to a name" $
             ifIndexToName 1 `shouldReturn` Just lpdevname
 
+    describe "socket" $ do
+        let gc = do
+                threadDelay 100000
+                performGC
+            connect' = do
+                threadDelay 200000
+                sock <- socket AF_INET Stream defaultProtocol
+                connect sock $ SockAddrInet 6000 $ tupleToHostAddress (127, 0, 0, 1)
+        it "can be GCed but not GCed when blocking" $ do
```

Comment:
Does this test that it gets GC'd? I thought what we want to make sure is that it does not get GC'd.

Reply:
Historically, `Socket` cannot be GCed. The major feature of version 3 is that `Socket` can be GCed. So, `Socket` should be GCed in proper time. Anyway, I will consider how to rephrase it.